docs/tutorial/go.rst

```diff
@@ -0,0 +1,361 @@
+    larger than the size of the compressed ``.rock`` file.
+
+Now we're finally ready to run the rock and test your containerised Go
+application:
+
+.. literalinclude:: code/go/task.yaml
+    :language: text
+    :start-after: [docs:docker-run]
+    :end-before: [docs:docker-run-end]
+    :dedent: 2
+
+Use the same ``curl`` command as before to send a request to the Go
+application's root endpoint which is running inside the container:
+
+.. literalinclude:: code/go/task.yaml
+    :language: text
+    :start-after: [docs:curl-go-rock]
+    :end-before: [docs:curl-go-rock-end]
+    :dedent: 2
+
+The Go application should again respond with ``Hello World``.
+
+
+View the application logs
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+When deploying the Go rock, you can always get the application logs via
+``pebble``:
+
+.. literalinclude:: code/go/task.yaml
+    :language: text
+    :start-after: [docs:get-logs]
+    :end-before: [docs:get-logs-end]
+    :dedent: 2
+
+As a result, :ref:`pebble_explanation_page` will give you the logs for the
+``go`` service running inside the container.
+You should expect to see something similar to this:
+
+..  code-block:: text
+    :class: log-snippets
+
+    2024-10-01T06:32:50.180Z [fastapi] INFO:     Started server process [12]
+    2024-10-01T06:32:50.181Z [fastapi] INFO:     Waiting for application startup.
+    2024-10-01T06:32:50.181Z [fastapi] INFO:     Application startup complete.
+    2024-10-01T06:32:50.182Z [fastapi] INFO:     Uvicorn running on http://0.0.0.0:8000 (Press CTRL+C to quit)
+    2024-10-01T06:32:58.214Z [fastapi] INFO:     172.17.0.1:55232 - "GET / HTTP/1.1" 200 OK
+
+You can also choose to follow the logs by using the ``-f`` option with the
+``pebble logs`` command above. To stop following the logs, press :kbd:`Ctrl` + :kbd:`C`.
+
+
+Cleanup
+~~~~~~~
+
+Now we have a fully functional rock for a Go application! This concludes
+the first part of this tutorial, so we'll stop the container and remove the
+respective image for now:
+
+.. literalinclude:: code/go/task.yaml
+    :language: bash
+    :start-after: [docs:stop-docker]
+    :end-before: [docs:stop-docker-end]
+    :dedent: 2
+
+
+Add another binary to the Go application
+========================================
+
+As a final step, let's update our application to include another binary.
+
+Create a new ``main.go`` file in the directory  ``cmd/anotherserver`` relative
+to the base directory of the project, with the following text and then
+save it:
+
+.. literalinclude:: code/go/cmd/anotherserver/main.go
+    :language: go
+
+Add the following snippet to the file ``rockcraft.yaml``, so the new binary is
+set as the main application to run:
+
+.. literalinclude:: code/go/anotherserver_part
+    :language: yaml
+
+Since we are creating a new version of the application, **open the**
+``rockcraft.yaml`` file and change the ``version`` (e.g. to ``0.2``).
+
+.. note::
+
+    ``rockcraft pack`` will create a new image with the updated code even if you
```

Comment:
Nitpick: Should this pack command have `ROCKCRAFT_ENABLE_EXPERIMENTAL_EXTENSIONS` included in it?

Reply:
Not sure about that, the command to run is just below this line (with the correct env var set). I think this is referring to the generic "rockcraft pack". What do you think?